src/clj_momo/lib/clj_time/core.cljc

```diff
@@ -0,0 +1,614 @@
+     (before? [this ^ReadablePartial that]
+       (time-delegate/before? this that))
+     (plus- [this ^ReadablePeriod period]
+       (time-delegate/plus- this period))
+     (minus- [this ^ReadablePeriod period]
+       (time-delegate/minus- this period))
+
+     org.joda.time.LocalTime
+     (hour [this] (time-delegate/hour this))
+     (minute [this] (time-delegate/minute this))
+     (second [this] (time-delegate/second this))
+     (milli [this] (time-delegate/milli this))
+     (equal? [this ^ReadablePartial that]
+       (time-delegate/equal? this that))
+     (after? [this ^ReadablePartial that]
+       (time-delegate/after? this that))
+     (before? [this ^ReadablePartial that]
+       (time-delegate/before? this that))
+     (plus- [this ^ReadablePeriod period]
+       (time-delegate/plus- this period))
+     (minus- [this ^ReadablePeriod period]
+       (time-delegate/minus- this period))
+
+     java.util.Date
+     (year [this]
+       (.get (doto (GregorianCalendar.)
+               (.setTimeZone TimeZoneUTC)
+               (.setTime this))
+             Calendar/YEAR))
+     (month [this]
+       (inc
+        (.get (doto (GregorianCalendar.)
+                (.setTimeZone TimeZoneUTC)
+                (.setTime this))
+              Calendar/MONTH)))
+     (day [this]
+       (.get (doto (GregorianCalendar.)
+               (.setTimeZone TimeZoneUTC)
+               (.setTime this))
+             Calendar/DAY_OF_MONTH))
+     (day-of-week [this]
+       (condp = (.get (doto (GregorianCalendar.)
+                        (.setTimeZone TimeZoneUTC)
+                        (.setTime this))
+                      Calendar/DAY_OF_WEEK_IN_MONTH)
+         Calendar/MONDAY DateTimeConstants/MONDAY
+         Calendar/TUESDAY DateTimeConstants/TUESDAY
+         Calendar/WEDNESDAY DateTimeConstants/WEDNESDAY
+         Calendar/THURSDAY DateTimeConstants/THURSDAY
+         Calendar/FRIDAY DateTimeConstants/FRIDAY
+         Calendar/SATURDAY DateTimeConstants/SATURDAY
+         Calendar/SUNDAY DateTimeConstants/SUNDAY))
+     (hour [this]
+       (.get (doto (GregorianCalendar.)
+               (.setTimeZone TimeZoneUTC)
+               (.setTime this))
+             Calendar/HOUR_OF_DAY))
+     (minute [this]
+       (.get (doto (GregorianCalendar.)
+               (.setTimeZone TimeZoneUTC)
+               (.setTime this))
+             Calendar/MINUTE))
+     (second [this]
+       (.get (doto (GregorianCalendar.)
+               (.setTimeZone TimeZoneUTC)
+               (.setTime this))
+             Calendar/SECOND))
+     (milli [this]
+       (.get (doto (GregorianCalendar.)
+               (.setTimeZone TimeZoneUTC)
+               (.setTime this))
+             Calendar/MILLISECOND))
+     (equal? [this that]
+       (.equals this that))
+     (after? [this that]
+       (< 0 (.compareTo this that)))
+     (before? [this that]
+       (> 0 (.compareTo this that)))
+     (plus- [this ^ReadableInstant period]
```

Comment:
What about refactoring these in a similar way to above?
```clojure
(defn time-op [^ Date date op &args]
  (Date. (.getMillis (apply op (time-coerce/to-date-time date) args))))
```

So `plus-` is:
`(plus- [this ^ReadableInstance period] (time-op this plus- period))`
And `first-day-of-the-month-` is:
`(plus- [this ^ReadableInstance period] (time-op this first-day-of-the-month-))`